File: Php/core/Navigation/NavigationNode.php
namespace core\Navigation{

	use app\Urls\BaseClasses\UrlModel;

	class NavigationNode{

		public $name;
        public $urlModel;
		public $selected;

		public function __construct($name, UrlModel $urlModel, $selected = false){
			name = $name;
			urlModel = $urlModel;
			selected = $selected;
		}

		public function getName(){
			return name;
		}

		public function getUrl(){
			return urlModel;
		}

		public function isSelected(){
			return selected;
		}

	}

}

File: Web/Models/NavigationNode.cs
namespace Web.Models{

	public class NavigationNode{

	    public string Name { get; private set; }
	    public UrlModel UrlModel { get; private set; }
	    public bool Selected { get; private set; }

	    public NavigationNode(string name, UrlModel urlModel, bool selected)
	    {
	        Name = name;
	        UrlModel = urlModel;
	        Selected = selected;
	    }

	}

}